### PR TITLE
Hooked up post endpoints.

### PR DESCRIPTION
**Pull request by Sashimi4**

### CODE REVIEW

**Comment by Sashimi4** [comment]
Left the image in the post out for the time being.

File: backend/src/main/java/com/social/dailylink/controller/PostController.java [hunk 1]
```diff
@@ -27,4 +31,33 @@ public ResponseEntity<Void> deleteById(@PathVariable String id) {
...
+    @Override
+    @PostMapping("/{id}")
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<PostDTO> create(@RequestBody PostDTO dto) {
+        return super.create(dto);
+    }
+
+    @Override
+    @PutMapping("/{id}")
+    //preAuthorize : Leaving it currently just for ADMINS.
```
Comment (davidemarcoli) [inline backend/src/main/java/com/social/dailylink/controller/PostController.java:58]: Discussions needed about this topic. But for the Moment it's okay!

File: backend/src/main/java/com/social/dailylink/service/PostServiceImpl.java [hunk 2]
```diff
@@ -26,10 +28,6 @@ public PostServiceImpl(AbstractEntityRepository<Post> repository, UserRepository
 
     @Override
     public Post create(Post entity) {
```
Comment (davidemarcoli) [inline backend/src/main/java/com/social/dailylink/service/PostServiceImpl.java:30]: There is a preSave Method in the AbstractEntityService for this purpose

File: backend/src/main/java/com/social/dailylink/service/PostServiceImpl.java [hunk 3]
```diff
@@ -40,11 +38,12 @@ public Post create(Post entity) {
     }
 
     @Override
-    public Post save(Post entity) {
```
Comment (davidemarcoli) [inline backend/src/main/java/com/social/dailylink/service/PostServiceImpl.java:43]: Same as the create() Method

File: backend/src/main/java/com/social/dailylink/service/PostServiceImpl.java [hunk 4]
```diff
@@ -40,11 +38,12 @@ public Post create(Post entity) {
...
-        if (((PostRepository) repository).findByTitle(entity.getTitle()).isPresent()) {
-            throw new EntityAlreadyExistsException("Post " + entity.getTitle() + " already exists");
-        }
+    public List<Post> findAll() {
```
Comment (davidemarcoli) [inline backend/src/main/java/com/social/dailylink/service/PostServiceImpl.java:35]: You only have to override the Methods you want to make an own implementation of. The others are automatically "generated".

**Review by davidemarcoli — CHANGES_REQUESTED** [review]

File: backend/src/main/java/com/social/dailylink/service/PostServiceImpl.java (hunk 2 above)
Comment (Sashimi4) [reply backend/src/main/java/com/social/dailylink/service/PostServiceImpl.java:30]: Ich han kei ahnig was presave macht 😀

Comment (davidemarcoli) [reply backend/src/main/java/com/social/dailylink/service/PostServiceImpl.java:30]: Das wird immer vor em Save usgfüert. Chasch ja mal d'Implementation vo save im AbstractEntityServiceImpl aluege

**Review by davidemarcoli — APPROVED** [review]
LGTM!

**Review by lazarpe — APPROVED** [review]

**Pull request merged**